toolkit/tools/imagegen/configuration/networkconfig.go

```diff
@@ -0,0 +1,309 @@
+		if err != nil {
+			return err
+		} else if deviceName == "" {
+			err = fmt.Errorf("The input network device is currently not supported on this system")
+			return err
+		}
+
+		err = createNetworkConfigFile(installChroot, networkData, deviceName)
+		if err != nil {
+			return err
+		}
+
+		dnsUpdate = len(networkData.NameServers) > 0
+	}
+
+	err = shell.ExecuteLive(squashErrors, "systemctl", "restart", "systemd-networkd")
+	if err != nil {
+		logger.Log.Errorf("Unable to restart systemd-networkd: %s", err)
+		return
+	}
+
+	if dnsUpdate {
+		err = shell.ExecuteLive(squashErrors, "systemctl", "restart", "systemd-resolved")
+		if err != nil {
+			logger.Log.Errorf("Unable to restart systemd-resolved: %s", err)
+		}
+	}
+
+	return
+}
+
+// bootProtoIsValid returns an error if input bootproto is invalid
+func (n *Network) bootProtoIsValid() (err error) {
+	n.BootProto = strings.TrimSpace(n.BootProto)
+	if validBootProtos[n.BootProto] {
+		return
+	}
+
+	return fmt.Errorf("invalid input for bootproto (%s), bootproto can only be one of dhcp, bootp, ibft and static", n.BootProto)
+}
+
+func (n *Network) validateIPAddress(ip string) (err error) {
+	if ip == "" && (n.BootProto == "" || n.BootProto == "dhcp") {
+		return
+	} else if net.ParseIP(ip) == nil {
+		return fmt.Errorf("invalid ip address (%s)", ip)
+	}
+
+	return
+}
+
+// ipAddrIsValid returns an error if ip, gateway, netmask or nameserver inputs are invalid ip addresses
+func (n *Network) ipAddrIsValid() (err error) {
+	if err = n.validateIPAddress(n.Ip); err != nil {
+		return fmt.Errorf("invalid input for ip: %w", err)
+	}
+
+	if err = n.validateIPAddress(n.NetMask); err != nil {
+		return fmt.Errorf("invalid input for netmask: %w", err)
+	}
+
+	if err = n.validateIPAddress(n.GateWay); err != nil {
+		return fmt.Errorf("invalid input for gateway: %w", err)
+	}
+
+	for _, nameserver := range n.NameServers {
+		if err = n.validateIPAddress(nameserver); err != nil {
+			return fmt.Errorf("invalid input for nameserver: %w", err)
+		}
+	}
+
+	return
+}
+
+// deviceIsValid returns an error if the Device name is empty
+func (n *Network) deviceIsValid() (err error) {
+	n.Device = strings.TrimSpace(n.Device)
+	if n.Device == "" {
+		return fmt.Errorf("invalid input for device, device cannot be empty")
+	}
+	return
+}
+
+func findBootIfValue() (deviceAddr string, err error) {
+	const macAddressStartIndex = 3
+
+	bootifValue, ferr := GetKernelCmdLineValue("BOOTIF")
+	bootifValue = strings.TrimSpace(bootifValue)
+	if ferr != nil || bootifValue == "" {
+		err = ferr
+		return
+	}
+
+	// The bootif value in the cmdline set by pxelinux is of the following format:
+	// bootif=01-MAC Address, where each byte value of the MAC address is separated
+	// by dashes instead of colons. Therefore, we're reading from the 10th spot of the
+	// string to obtain the MAC address and then replace the dashes with colons
+	deviceAddr = strings.Replace(bootifValue[macAddressStartIndex:len(bootifValue)], "-", ":", -1)
```

Comment:
Suggestion: we can drop the last argument if we use `strings.ReplaceAll` instead.

Reply:
Good point. Fixed